esphome/components/atm90e32/atm90e32.cpp

```diff
@@ -262,26 +403,35 @@ float ATM90E32Component::get_setup_priority() const { return setup_priority::IO;
 // R/C registers can conly be cleared after the LastSPIData register is updated (register 78H)
 // Peakdetect period: 05H. Bit 15:8 are PeakDet_period in ms. 7:0 are Sag_period
 // Default is 143FH (20ms, 63ms)
-uint16_t ATM90E32Component::read16_(uint16_t a_register) {
+uint16_t ATM90E32Component::read16_transaction_(uint16_t a_register) {
   uint8_t addrh = (1 << 7) | ((a_register >> 8) & 0x03);
   uint8_t addrl = (a_register & 0xFF);
-  uint8_t data[2];
-  uint16_t output;
+  uint8_t data[4] = {addrh, addrl, 0x00, 0x00};
+  this->transfer_array(data, 4);
+  uint16_t output = (uint16_t(data[2] & 0xFF) << 8) | (data[3] & 0xFF);
```

Comment:
We have a helper for this with casting
```suggestion
  uint16_t output = encode_uint16(data[2], data[3]);
```